feat(SPV-1592): create engine v2

dorzepowski opened the pull request:
<!-- 
    Thank you for contributing to our project! Before you submit your Pull Request, please make sure you've completed the items in this checklist.
    Please check the boxes below by putting an x in the [ ] like so: [x]. You can do it right after creating the PR.
-->

# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

<!-- 
## PR Title as Conventional Commit

Your PR title should also be a [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/). This will assist us in our release process and in assigning semantic version numbers to releases.
-->

<!-- 
## Related Tickets & Documents

If your changes relate to or close an issue, please include them below. We like to follow [Github's guidance on linking issues to pull requests](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

For instance, having the text "closes #1234" would link the current pull request to issue number 1234. And when we merge the pull request, Github will automatically close the issue.

- Related Issue #
- Closes #
-->

<!--
## Feature/Issue Details

Your pull request should be tightly connected to a feature or issue. Ensure that you've covered all necessary pathways and there's a possibility to test code behavior following the issue description or feature specification.
-->

dorzepowski committed:
feat(SPV-1592): create engine.V2

dorzepowski committed:
feat(SPV-1592): wrap engine.V2 into engine V1 as temporary adapting.

chris-4chain commented on engine/v2/utils/must/must.go:
```diff
@@ -0,0 +1,30 @@
+func HaveNoErrorf(err error, format string, args ...interface{}) {
+	msg := fmt.Sprintf(format, args...)
+	if err != nil {
+		panic(fmt.Errorf("%s, caused by: %w", msg, err))
```
Or `Fatal`?

dorzepowski replied:
Seems to me that panic could be more flexible choice for future.

chris-4chain commented on engine/v2/utils/must/must.go:
```diff
@@ -0,0 +1,30 @@
+// BeTrue is a simple way to panic if the condition is false.
+func BeTrue(condition bool, format string, args ...interface{}) {
+	if !condition {
+		panic(fmt.Sprintf(format, args...))
```
Or `Fatal`?

dorzepowski replied:
Seems that panic could be more flexible choice for future.

chris-4chain commented on engine/v2/engine/engine.go:
```diff
@@ -0,0 +1,258 @@
+	if cfg.ARC.Callback.Enabled {
+		var err error
+		if cfg.ARC.Callback.Token == "" {
+			// This also sets the token to the config reference and, it is used in the callbacktoken_middleware
+			// TODO: consider moving config modification to a PostLoad method and make this ToEngineOptions pure (no side effects)
+			cfg.ARC.Callback.Token, err = utils.HashAdler32(config.DefaultAdminXpub)
+			must.HaveNoErrorf(err, "error while generating callback token")
```
Maybe this TODO is no longer applicable. Maybe it's ok that we create the token here.

dorzepowski replied:
IMHO it should be part of the config initialisation, but we should revisit config again. Maybe the best would be to create a task for that.

chris-4chain commented on engine/v2/engine/engine_overrides.go:
```diff
@@ -0,0 +1,42 @@
+	}
+}
+
+// WithTransport is a function that can be used to override the http.RoundTripper used by the engine.
+// This is meant to be used for testing purposes.
+func WithTransport(transport http.RoundTripper) InternalsOverride {
+	return func(o *overrides) {
+		o.transport = transport
+	}
+}
```
This seems to be unused

dorzepowski commented on engine/v2/engine/internal/storage.go:
```diff
@@ -0,0 +1,176 @@
+// CreateUTXOSelector creates a new instance of the UTXO selector.
+func (s *Storage) CreateUTXOSelector(feeService *fee.Service) outlines.UTXOSelector {
+	must.BeTrue(s.db != nil, "Trying to create utxo selector on closed database connection")
+	// TODO: pass feeService instead of simply fee
```
I will create a SPV-1594

dorzepowski commented on actions/v2/callback/callbacks.go:
```diff
@@ -0,0 +1,51 @@
+func RegisterRoutes(ginEngine *gin.Engine, cfg *config.AppConfig, engine engine.V2Interface) {
+	if cfg.ARCCallbackEnabled() {
+		callbackURL, err := cfg.ARC.Callback.ShouldGetURL()
+		must.HaveNoErrorf(err, "couldn't get callback URL from configuration")
```
thx

dorzepowski committed:
chore(SPV-1592): remove todo comments, because there are tasks for them.

mrz1836 committed:
Merge branch 'main' into engine-v2

Resolved merge conflicts by integrating main's new features into engine-v2 architecture:

- Added contacts service to engine.V2 (from SPV-1538)
- Integrated errorx & RFC 7807 problem details (from SPV-1583)
- Incorporated admin record transaction endpoint (from SPV-1544)
- Included user create address endpoint (from SPV-1548)

Resolution strategy:
- Preserved engine-v2's unified engine.V2 abstraction
- Updated V2Interface to include ContactService() method
- Modified paymail service provider to accept contacts parameter
- Removed individual v2 service loaders (handled by engine.V2)

All conflicts resolved while maintaining architectural consistency.

mrz1836 committed:
feat(errors): add SPVError legacy interface support

Handle legacy SPVError types in error conversion to problem details format

mrz1836 committed:
test(contacts): update assertions to use WithProblemDetails

Migrate test assertions from HasStatus/WithJSONf pattern to new WithProblemDetails helper method
